pkg/piecestore/pstore.go

```diff
@@ -20,86 +19,70 @@ var (
 	FSError  = errs.Class("fsError")
 )
 
-// PathByHash -- creates datapath from hash and dir
-func PathByHash(hash, dir string) (string, error) {
-	if len(hash) < 20 {
-		return "", ArgError.New("Invalid hash length")
+// PathByID -- creates datapath from id and dir
+func PathByID(id, dir string) (string, error) {
+	if len(id) < 20 {
+		return "", ArgError.New("Invalid id length")
+	}
+	if dir == "" {
+		return "", ArgError.New("No path provided")
 	}
 
-	folder1 := string(hash[0:2])
-	folder2 := string(hash[2:4])
-	fileName := string(hash[4:])
+	folder1 := string(id[0:2])
```

Comment:
what's the format of id ? Would it be possible for this to change in the future?

Reply:
Id is 5+ ascii letters